src/pelicanfs/core.py

```diff
@@ -47,6 +48,50 @@ class InvalidMetadata(PelicanException):
     No Pelican metadata was found for the federation
     """
 
+class _AccessResp:
+    def __init__(self, path: str, success: bool, error: Optional[str] = None):
+        self.access_path = path
+        self.success = success
+        self.error = error
+    
+    def __repr__(self) -> str:
+        if self.error:
+            return f"{{NamespacePath: {self.access_path}, Success: {self.success}, Error: {self.error}}}"
+        return f"{{NamespacePath: {self.access_path}, Success: {self.success}}}"
+    
+class _AccessStats:
+    def __init__(self):
+        """
+        Manage the cache access stats
+
+        For each namespace path, keep a list of the last three cache responses, including the
+        cache path, a boolean which is try if successful, and an error string if not
```

Comment:
What's a cache path in this context? Is it a URL? I'm also a bit confused about what you mean about "a boolean which is try if successful".